integration/tests/cook/test_basic.py

```diff
@@ -965,6 +995,66 @@ def all_completed_restarted(response):
             # ensure that we don't leave a bunch of jobs running/waiting
             util.kill_groups(self.cook_url, [group_uuid])
 
+    def _help_test_group_failed_only_change_retries(self, config):
```

Comment:
To me, this is hard to understand without reading the corresponding tests and jumping back and forth. How long would it be if you just had 3 separate tests (assuming there's some generic code in here that not all tests need) You could probably kill a lot of the comments in that case as well, because the code would be more straightforward.

Reply:
I agree that the generic `config` arg makes this harder to understand, but I also don't want to duplicate all of this code three times. I don't see a winning solution here... We'll have to defer to @dposada to decide which is the lesser evil in this case.

Reply:
I am generally against having test assertions not be in the test proper (i.e. in a helper function). How about keeping the meat of this function, but moving the assertions back out to each individual test?

Reply:
That seems like a good compromise.

Reply:
Done! Lifting the test assertions out of the helper method also simplified the config map I was passing to the helper. I think this change was definitely an improvement. Thanks for the feedback!